--- a/readers/zpdfreader.cpp
+++ b/readers/zpdfreader.cpp
@@ -93,7 +93,7 @@ bool ZPdfReader::openFile()
         return false;
     }
 
-    GooString fname(fileName.toUtf8());
+    GooString fname(fileName.toUtf8().constData());
     m_doc = PDFDocFactory().createPDFDoc(fname);
 
     if (m_doc==nullptr || !m_doc->isOk())
@@ -263,7 +263,7 @@ void ZPdfReader::loadPagePrivate(int idx, QByteArray *data, QImage *image, bool
         const quint8 cBlue = 0xff;
         const quint8 cAlpha = 0;
         SplashColor bgColor { cRed, cGreen, cBlue, cAlpha };
-        SplashOutputDev splashOutputDev(splashModeXBGR8, 4, false, bgColor, true);
+        SplashOutputDev splashOutputDev(splashModeXBGR8, 4, bgColor, true);
         splashOutputDev.setFontAntialias(true);
         splashOutputDev.setVectorAntialias(true);
         splashOutputDev.setFreeTypeHinting(true, false);
