--- Core/FileIO/NBFileIO.cpp	2014-02-13 14:03:01.000000000 +0400
+++ Core/FileIO/NBFileIO.cpp	2014-02-20 14:21:49.881703784 +0400
@@ -261,7 +261,7 @@
 	}
 
 	int iFileFD = open( qPrintable( srcFile ), O_RDONLY );
-	int oFileFD = open( qPrintable( ioTarget ), O_WRONLY | O_CREAT );
+	int oFileFD = open( qPrintable( ioTarget ), O_WRONLY | S_IWUSR | O_CREAT );
 
 	fTotalBytes = iStat.st_size;
 	fWritten = 0;
