--- crengine/src/old-lvfntman.cpp
+++ crengine/src/lvfntman.cpp
@@ -43,11 +43,11 @@
 //#include <ft2build.h>
 
 #ifdef ANDROID
-#include "freetype/config/ftheader.h"
-#include "freetype/freetype.h"
+#include "freetype2/config/ftheader.h"
+#include "freetype2/freetype.h"
 #else
 
-#include <freetype/config/ftheader.h>
+#include <freetype2/config/ftheader.h>
 //#include <ft2build.h>
 #include FT_FREETYPE_H
 //#include <freetype/freetype.h>
@@ -3996,4 +3996,3 @@
             && r1.getHintingMode()==r2.getHintingMode()
             ;
 }
-
