diff --git a/portconf b/portconf
index f1253d4..53c7289 100755
--- a/portconf
+++ b/portconf
@@ -26,8 +26,8 @@ ARCH="$(eix --print DEFAULT_ARCH)"
 COUNT="${COUNT:-10}"
 ############################################################################
 ################ ignore ####################################################
-[[ -n "${IGNORE_CATEGORY}" ]] && IGNORE_CATEGORY="$(sed -e '/^$/d' -e 's/$/\/.*/g' <<< "`tr "[:space:]" $'\n' <<< "${IGNORE_CATEGORY}"`")"
-[[ -n "${IGNORE_PN}" ]] && IGNORE_PN="$(sed -e '/^$/d' -e 's/^/.*\//g' <<< "`tr "[:space:]" $'\n' <<< "${IGNORE_PN}"`")"
+[[ -n "${IGNORE_CATEGORY}" ]] && IGNORE_CATEGORY="$(sed -e '/^$/d' -e 's/$/\/.*/g' <<< "$(tr "[:space:]" $'\n' <<< "${IGNORE_CATEGORY}")")"
+[[ -n "${IGNORE_PN}" ]] && IGNORE_PN="$(sed -e '/^$/d' -e 's/^/.*\//g' <<< "$(tr "[:space:]" $'\n' <<< "${IGNORE_PN}")")"
 IGNORE="${IGNORE_CATEGORY}
 ${IGNORE_PN}"
 IGNORE="$(sed -e '/^$/d' <<< "${IGNORE}")"
@@ -37,11 +37,11 @@ IGNORE="${IGNORE:-'Hello, LOR! :3'}"
 sort_passed_uses(){
 	for opt in ${1};do
 		opt="${opt//\*/\\*}"
-		uses="$(tr "[:space:]" $'\n' <<< ${uses} | grep -v -e "^${opt#-}$" -e "^\-${opt#-}$" | tr $'\n' " ")"
-		uses+=" $(tr "[:space:]" $'\n' <<< ${1} | grep -e "^${opt#-}$" -e "^\-${opt#-}$" | tail -n1)"
+		uses="$(tr "[:space:]" $'\n' <<< "${uses}" | grep -v -e "^${opt#-}$" -e "^-${opt#-}$" | tr $'\n' " ")"
+		uses+=" $(tr "[:space:]" $'\n' <<< "${1}" | grep -e "^${opt#-}$" -e "^-${opt#-}$" | tail -n1)"
 		unset opt
 	done
-	echo ${uses}
+	echo "${uses}"
 	unset uses
 }
 ############################################################################
@@ -51,35 +51,35 @@ profile() {
 	[[ -f "${1}/make.defaults" ]] && . "${1}/make.defaults"
 	use="$USE"
 	if [[ -f "${1}/parent" ]]; then
-		for path in `cat ${1}/parent`;do
+		for path in $(cat "${1}"/parent);do
 			use="$use $(profile "${1}/$path")"
 		done
 	fi
-	echo $use
+	echo "$use"
 }
 
-[[ -h "/etc/make.profile" ]] && pre_profile="$(profile `readlink -f /etc/make.profile`)"
-[[ -h "/etc/portage/make.profile" ]] && pre_profile="$(profile `readlink -f /etc/portage/make.profile`)"
+[[ -h "/etc/make.profile" ]] && pre_profile="$(profile $(readlink -f /etc/make.profile))"
+[[ -h "/etc/portage/make.profile" ]] && pre_profile="$(profile $(readlink -f /etc/portage/make.profile))"
 profile_uses="$(sort_passed_uses "${pre_profile}")"
 PROFILE="$(tr "[:space:]" $'\n' <<< ${profile_uses})"
 ############################################################################
 ################ all_use_all_profiles ######################################
 all_use_all_profiles(){
-	for target in `eselect --brief profile list`;do
+	for target in $(eselect --brief profile list);do
 		profile_uses="$(profile "${PORTDIR}/profiles/${target}")"
 		USEs="$(sort_passed_uses "${profile_uses}")"
-		echo -e "\n${green}${target}${restore}: "${USEs} | fold -s -w `tput cols`
+		echo -e "\n${green}${target}${restore}: "${USEs} | fold -s -w $(tput cols)
 		unset USEs
 	done
 }
 ############################################################################
 ################ profile_use ###############################################
 profile_use(){
-	PS3=`echo -e "${contrast}Choose number:${restore} "`
-	select target in `eselect --brief profile list`;do
+	PS3=$(echo -e "${contrast}Choose number:${restore} ")
+	select target in $(eselect --brief profile list);do
 		profile_uses="$(profile "${PORTDIR}/profiles/${target}")"
 		USEs="$(sort_passed_uses "${profile_uses}")"
-		echo -e "\n${green}${target}${restore}: "${USEs} | fold -s -w `tput cols`
+		echo -e "\n${green}${target}${restore}: "${USEs} | fold -s -w $(tput cols)
 		unset USEs
 		break
 	done
@@ -94,7 +94,7 @@ current_profile(){
 	fi
 	profile_uses="$(profile "${PORTDIR}/profiles/${target}")"
 	USEs="$(sort_passed_uses "${PROFILE}")"
-	echo -e "${green}${target}${restore}: "${USEs} | fold -s -w `tput cols`
+	echo -e "${green}${target}${restore}: "${USEs} | fold -s -w $(tput cols)
 	unset USEs
 }
 ############################################################################
@@ -106,6 +106,10 @@ MAKE_USES="${USE}"
 unset USE
 PORTDIR="${PORTDIR:-/usr/portage}"
 ############################################################################
+################ temp directory ##########################################
+TMPDIR_PORTCONF="$(mktemp -d)"
+############################################################################
+############################################################################
 ################ timestamp #################################################
 timestamp(){
 	stat -c %Y ${*} 2>/dev/null | sort | tail -n1
@@ -143,14 +147,13 @@ eix_check(){
 	if [[ -n "$(eix --version 2>/dev/null)" ]];then
 		echo -e "${underline}Creating new temporary cache.${restore}"
 		repo_fix
-		eix_cache="$(mktemp)"
+		eix_cache="${TMPDIR_PORTCONF}/eix_cache"
 		chown portage:portage "${eix_cache}" 2>/dev/null
 		export EIX_CACHEFILE="${eix_cache}"
 		export OVERLAY_CACHE_METHOD="parse|ebuild*"
 		eix-update
 	else
 		echo -e "${green}app-portage/eix${restore} ${red}not installed!${restore}"
-		eend 1 2>/dev/null
 		exit 1
 	fi
 	rm_repo_fix
@@ -177,7 +180,6 @@ Create temporary cache?${restore} [${green}Yes${restore}/${red}No${restore}] ";
 		fi
 	else
 		echo -e "${green}app-portage/eix${restore} ${red}not installed!${restore}"
-		eend 1 2>/dev/null
 		exit 1
 	fi
 }
@@ -208,10 +210,8 @@ backup(){
 		else
 			tar -jcf "${name}" -C /etc/ portage
 		fi
-		eend $? 2>/dev/null
 	else
 		echo -e "/etc/portage backups: ${green}already up-to-date${restore}"
-		eend $? 2>/dev/null
 	fi
 }
 ############################################################################
@@ -234,10 +234,8 @@ world_backup(){
 		echo -e "world backup: ${green}creating${restore}..."
 		name="${world_dir}/world_$(date +%g.%m.%d-%H:%M).tar.bz2"
 		tar -jcf "${name}" -C ${WORLD%world} world
-		eend $? 2>/dev/null
 	else
 		echo -e "world backups: ${green}already up-to-date${restore}"
-		eend $? 2>/dev/null
 	fi
 }
 ############################################################################
@@ -247,13 +245,12 @@ restore(){
 	tarballs="${tarballs//${1}\/}"
 	tarballs="${tarballs//.tar.bz2/}"
 	tarballs="${tarballs//${2}_/}"
-	PS3=`echo -e "${contrast}Choose number:${restore} "`
+	PS3=$(echo -e "${contrast}Choose number:${restore} ")
 	echo -e "${contrast}Available backups${restore}:"
 	select answer in ${tarballs};do
 		echo -e "Rolling back ${3} to the state ${green}${answer}${restore}"
-		[[ -n "${5}" ]] && rm -rf ${5}
+		[[ -n "${5}" ]] && rm -rf "${5}"
 		tar -xf "${1}/${2}_${answer}.tar.bz2" -C ${4} 2>/dev/null
-		eend $? 2>/dev/null
 		break
 	done
 }
@@ -281,18 +278,18 @@ diff_ask(){
 			action="$(grep -o "[a-z]" <<< "${subj}")"
  			if [[ "${action}" == "a" ]];then
 				for num in ${new_num};do
-					[[ "${num}" != "0" ]] && full_diff+="${num}^#+#${green}`sed -n "${num}p" "${2}"`${restore}#"$'\n'
+					[[ "${num}" != "0" ]] && full_diff+="${num}^#+#${green}$(sed -n "${num}p" "${2}")${restore}#"$'\n'
 				done
 			elif [[ "${action}" == "d" ]];then
 				for num in ${old_num};do
-					[[ "${num}" != "0" ]] && full_diff+="${num}^#-#${red}`sed -n "${num}p" "${1}"`${restore}#"$'\n'
+					[[ "${num}" != "0" ]] && full_diff+="${num}^#-#${red}$(sed -n "${num}p" "${1}")${restore}#"$'\n'
 				done
 			elif [[ "${action}" == "c" ]];then
 				for num in ${old_num};do
-					[[ "${num}" != "0" ]] && full_diff+="${num}^#-#${red}`sed -n "${num}p" "${1}"`${restore}#"$'\n'
+					[[ "${num}" != "0" ]] && full_diff+="${num}^#-#${red}$(sed -n "${num}p" "${1}")${restore}#"$'\n'
 				done
 				for num in ${new_num};do
-					[[ "${num}" != "0" ]] && full_diff+="${num}^#+#${green}`sed -n "${num}p" "${2}"`${restore}#"$'\n'
+					[[ "${num}" != "0" ]] && full_diff+="${num}^#+#${green}$(sed -n "${num}p" "${2}")${restore}#"$'\n'
 				done
 			fi
 		done <<< "${my_diff}"
@@ -333,7 +330,7 @@ diff_ask(){
 			unset full_diff
 			rm "${2}" 2>/dev/null
 			return 0
-			break
+				# break removed - dead code after return 0 above
 		fi
 	fi
 	rm "${2}" 2>/dev/null
@@ -347,7 +344,7 @@ remove_ask(){
 		while read x;do
 			case ${x} in
 				Yes | YES | yes | Y | y | "" )
-					rm -rf ${1} 2>/dev/null
+					rm -rf "${1}" 2>/dev/null
 					break;;
 				No | NO | no | N | n )
 					break;;
@@ -355,10 +352,9 @@ remove_ask(){
 					continue;;
 			esac
 		done
-		eend $? 2>/dev/null
 	elif [[ -n "${yes}" && -z "${PRETEND}" ]];then
 		echo -e "${contrast}Removing${restore} ${green}${1}${restore}"
-		rm -rf ${1} 2>/dev/null
+		rm -rf "${1}" 2>/dev/null
 		eend $? 2>/dev/null
 	elif [[ -z "${yes}" && -n "${PRETEND}" ]];then
 		eend 0 2>/dev/null
@@ -367,20 +363,19 @@ remove_ask(){
 ############################################################################
 ################ sort_packages #############################################
 sort_uniq_files(){
-	for package in `find ${PORT_ETC}/ -type f \! -name '*~' \! -name '*.bak' | grep "/package\."`;do
-		tmp_sort="$(mktemp)"
+	for package in $(find ${PORT_ETC}/ -type f \! -name '*~' \! -name '*.bak' | grep "/package\.");do
+		tmp_sort="${TMPDIR_PORTCONF}/tmp_sort"
 		echo -e "${underline}Sorting${restore} ${contrast}${package#${PORT_ETC}/}${restore}"
 
-		all_atoms="$(awk '{print $1}' "${package}" | grep -v "^#"| sort -u)"
+		all_atoms="$(awk '!/^#/ {print $1}' "${package}"| sort -u)"
 		while read atom;do
 			opts="$(grep "^${atom//\*/\\*} " "${package}" | uniq | cut -d" " -f2-)"
 			echo "${atom} ${opts//$'\n'/ }" >> "${tmp_sort}"
 		done <<< "${all_atoms}"
 
 		sed -e '/^$/d' -e 's/[ \t]*$//' -i ${tmp_sort}
-		sort -u "${tmp_sort}" -o "${tmp_sort}" 2>/dev/null
-		diff_ask "${package}" "${tmp_sort}"
-		eend $? 2>/dev/null
+		sort -u "${tmp_sort}" 2>/dev/null
+		[[-z "${NO_DIFF}"]] && diff_ask "${package}" "${tmp_sort}"
 	done
 }
 ############################################################################
@@ -392,7 +387,6 @@ file_or_dir(){
 	done
 	if [[ -z "${founded}" ]];then
 		echo -e "${underline}${2}${restore}${contrast}${PORT_ETC}/${1}${restore}: No such file or directory${restore}"
-		eend 1 2>/dev/null
 		return 1
 	fi
 	for target in ${founded};do
@@ -404,10 +398,9 @@ file_or_dir(){
 			else
 				echo -e "${underline}Empty file${restore} ${contrast}${file}${restore}"
 				remove_ask "${file}"
-				eend $? 2>/dev/null
 			fi
 		elif [[ -d "${target}" ]];then
-			for f in `find ${target} -type f \! -name '*~' \! -name '*.bak'`;do
+			for f in $(find ${target} -type f \! -name '*~' \! -name '*.bak');do
 				file="${f}"
 				if [[ -s ${file} ]];then
 					echo -e "${underline}${2}${restore} ${contrast}${file#${PORT_ETC}/}${restore}"
@@ -415,7 +408,6 @@ file_or_dir(){
 				else
 					echo -e "${underline}Empty file${restore} ${contrast}${file}${restore}"
 					remove_ask "${file}"
-					eend $? 2>/dev/null
 					continue
 				fi
 			done
@@ -426,11 +418,10 @@ file_or_dir(){
 ################ rm_comments ###############################################
 rm_comments(){
 	no_newline(){
-		tmp_file="$(mktemp)"
+		tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 		cp "${file}" "${tmp_file}"
 		sed -e "s|^#.*||g" -e '/^$/d' -i ${tmp_file}
-		diff_ask "${file}" "${tmp_file}"
-		eend $? 2>/dev/null
+		[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${tmp_file}"
 	}
 	file_or_dir "package.*" "Checking comments:" "no_newline" || return 1
 }
@@ -438,20 +429,19 @@ rm_comments(){
 ################ rm_all_comments ###########################################
 rm_all_comments(){
 	no_newline_inline(){
-		tmp_file="$(mktemp)"
+		tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 		cp "${file}" "${tmp_file}"
 		sed -e "s|#.*||g" -e 's/[ \t]*$//' -e '/^$/d' -i ${tmp_file}
-		diff_ask "${file}" "${tmp_file}"
-		eend $? 2>/dev/null
+		[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${tmp_file}"
 	}
 	file_or_dir "package.*" "Checking comments:" "no_newline_inline"  || return 1
 }
 ############################################################################
 ################ sort_uses #################################################
 sort_uses(){
-	tmp_file="$(mktemp)"
-	striped="$(mktemp)"
-	PACKAGES="$(awk '{print $1}' ${file} | grep -v '^#.*' | uniq)"
+	tmp_file="${TMPDIR_PORTCONF}/tmp_file"
+	striped="${TMPDIR_PORTCONF}/striped"
+	PACKAGES="$(awk '!/^#/ {print $1}' ${file} | uniq)"
 	for atom in ${PACKAGES};do
 		comment="$(grep "${atom}[[:space:]].*#" "${file}" | cut -d\# -f2-)"
 		for option in \
@@ -474,10 +464,9 @@ sort_uses(){
 		unset atom opt
 	done < "${tmp_file}"
 	sed -e 's/[ \t]*$//' -e '/^$/d' -e 's|  | |g' -e "s| - | |g" -i "${striped}" 2>/dev/null
-	sort -u "${striped}" -o "${striped}" 2>/dev/null
+	sort -u "${striped}" 2>/dev/null
 	rm "${tmp_file}" 2>/dev/null
-	diff_ask "${file}" "${striped}"
-	eend $? 2>/dev/null
+	[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${striped}"
 } 2>/dev/null
 ############################################################################
 ################ package.use ###############################################
@@ -505,7 +494,7 @@ stupid_keywords(){
 					for pkey in ${parent_keys};do
 						if [[ "${key}" == "${KEY}" || ${key} == "${pkey}" ]];then
 							echo -e "Twice defined keyword ${red}${key}${restore}:\t${green}${satom}${restore}"
-							out_keys="$(grep -v "`printf ${key}`"  <<< "${out_keys}")"
+							out_keys="$(grep -v "$(printf ${key})"  <<< "${out_keys}")"
 						fi
 					done
 				done
@@ -533,9 +522,9 @@ stupid_keywords(){
 ############################################################################
 ################ sort_keys #################################################
 sort_keys(){
-	tmp_file="$(mktemp)"
-	striped="$(mktemp)"
-	PACKAGES="$(awk '{print $1}' ${file} | grep -v '^#.*' | uniq)"
+	tmp_file="${TMPDIR_PORTCONF}/tmp_file"
+	striped="${TMPDIR_PORTCONF}/striped"
+	PACKAGES="$(awk '!/^#/ {print $1}' ${file} | uniq)"
 	for atom in ${PACKAGES};do
 		comment="$(grep "${atom}[[:space:]].*#" "${file}" | cut -d\# -f2-)"
 		for option in \
@@ -560,19 +549,18 @@ sort_keys(){
 	done < "${tmp_file}"
 
 	sed -e 's/[ \t]*$//' -e '/^$/d' -e 's|  | |g' -e "s| - | |g" -i "${striped}" 2>/dev/null
-	sort -u "${striped}" -o "${striped}" 2>/dev/null
+	sort -u "${striped}" 2>/dev/null
 	cp "${striped}" "${tmp_file}"
 	stupid_keywords "${tmp_file}" "${striped}"
 	rm "${tmp_file}" 2>/dev/null
-	diff_ask "${file}" "${striped}"
-	eend $? 2>/dev/null
+	[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${striped}"
 } #2>/dev/null
 ############################################################################
 ################ one_keyword ###############################################
 sort_keywords(){
 	keys(){
-		tmp_file="$(mktemp)"
-		striped="$(mktemp)"
+		tmp_file="${TMPDIR_PORTCONF}/tmp_file"
+		striped="${TMPDIR_PORTCONF}/striped"
 		atoms="$(awk '{print $1}' "${file}" | sort -u)"
 		while read atom;do
 			comment="$(grep "${atom}[[:space:]].*#" "${file}" | cut -d\# -f2-)"
@@ -585,12 +573,11 @@ sort_keywords(){
 				echo "${atom} ${keyword}" >> "${tmp_file}"
 			fi
 		done <<< "${atoms}"
-		sort -u "${tmp_file}" -o "${tmp_file}" 2>/dev/null
+		sort -u "${tmp_file}" 2>/dev/null
 		cp "${tmp_file}" "${striped}"
 		stupid_keywords "${striped}" "${tmp_file}"
 		rm "${striped}"
-		diff_ask "${file}" "${tmp_file}"
-		eend $? 2>/dev/null
+		[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${tmp_file}"
 	}
 	file_or_dir "package.keywords package.accept_keywords" "Sorting keywords in" "keys" || return 1
 }
@@ -606,8 +593,8 @@ installed(){
 	if [[ -n "${installed}" ]];then
 		installed_vers="$(awk -F \: '{print $1}' <<< "${installed}")"
 		installed_min_ver="$(head -n1 <<< "${installed_vers}")"
-		installed_max_ver="$(tail -n1 <<< "`grep -v 9999 <<< "${installed_vers}"`")"
-		installed_live_ver="$(tail -n1 <<< "`grep 9999 <<< "${installed_vers}"`")"
+		installed_max_ver="$(tail -n1 <<< "$(grep -v 9999 <<< "${installed_vers}")")"
+		installed_live_ver="$(tail -n1 <<< "$(grep 9999 <<< "${installed_vers}")")"
 		installed_slots="$(awk -F \: '{print $2}' <<< "${installed}")"
 		#::repo check
 		full_vers="$(qlist -ICev "${category}/${pn}" | sed "s|${category}/${pn}-||g")"
@@ -624,7 +611,7 @@ installed(){
 }
 
 not_if(){
-	atoms="$(awk '{print $1}' <<< "`grep -v "^#" < ${package}`")"
+	atoms="$(awk '{print $1}' <<< "$(awk '!/^#/' < ${package})")"
 	[[ -n "${IGNORE}" ]] && atoms="$(grep -v "${IGNORE}" <<< "${atoms}")"
 	echo -e "${underline}Checking${restore} ${contrast}$package${restore}"
 	if [[ -n "${atoms}" ]];then
@@ -642,7 +629,7 @@ not_if(){
 
 				[[ "${category}" == "(null)" || "${pn}" == "(null)" ]] && continue
 
-				if [[ -n "${rev}" ]] && [[ -z "${slot}" ]] && `grep -q \: <<< "${rev}"`;then
+				if [[ -n "${rev}" ]] && [[ -z "${slot}" ]] && grep -q \: <<< "${rev}";then
 					slot="${rev}"
 					unset rev
 				fi
@@ -657,36 +644,32 @@ not_if(){
 						echo -e "${red}Not installed${restore} ${green}${category}/${pn}${restore}"
 					fi
 					sed -e "s|.*\<${category}/${pn}\>.*||" -e '/^$/d' -i "${tmp_package}" 2>/dev/null
-					eend $? 2>/dev/null
 				fi
-			done <<< `qatom "${atom}"`
+			done <<< $(qatom "${atom}")
 		done <<< "${atoms}"
 	fi
 }
 
 force_not_installed(){
-	for package in `find ${PORT_ETC}/package* -type f \! -name '*~' \! -name '*.bak'`;do
-		if [[ -s "${package}" && -n "$(grep -v "^#" < "${package}")" ]];then
-			tmp_package="$(mktemp)"
+	for package in $(find ${PORT_ETC}/package* -type f \! -name '*~' \! -name '*.bak');do
+		if [[ -s "${package}" && -n "$(awk '!/^#/' < "${package}")" ]];then
+			tmp_package="${TMPDIR_PORTCONF}/tmp_package"
 			cp "${package}" "${tmp_package}"
-			if ! `grep -q "\.mask" <<< "${package}"`;then
+			if ! grep -q "\.mask" <<< "${package}";then
 				not_if;
-				eend $? 2>/dev/null
 			fi
-			diff_ask "${package}" "${tmp_package}"
+			[[-z "${NO_DIFF}"]] && diff_ask "${package}" "${tmp_package}"
 		else
 			pack="${package%/*}"
 			echo -e "${underline}Empty file${restore} ${contrast}${pack}${restore}"
 			remove_ask "${package}"
-			eend $? 2>/dev/null
 		fi
 	done
 
-	for x in `find ${PORT_ETC}/package* -type f`;do
+	for x in $(find ${PORT_ETC}/package* -type f);do
 		if [[ ! -s "${x}" ]];then
 			echo -e "${underline}Empty file${restore} ${contrast}${x}${restore}"
 			remove_ask "${x}"
-			eend $? 2>/dev/null
 		fi
 	done
 }
@@ -708,7 +691,6 @@ package_env_conf(){
 			if ! grep -q "^${conf}$" <<< "${USED_ENV}";then
 				echo -e "Unused conf: ${red}${conf}${restore}"
 				remove_ask "${PORT_ETC}/env/${conf}"
-				eend $? 2>/dev/null
 			fi
 		done
 	}
@@ -718,7 +700,7 @@ package_env_conf(){
 ################ package.env ###############################################
 package_env(){
 	package_env_def(){
-		tmp_file="$(mktemp)"
+		tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 		cp "${file}" "${tmp_file}"
 		while read line;do
 			xline="$(sed 's/^[ \t]*//;s/[ \t]*$//' <<< "${line%%#*}")"
@@ -736,8 +718,7 @@ package_env(){
 				fi
 			done
 		done < "${file}"
-		diff_ask "${file}" "${tmp_file}"
-		eend $? 2>/dev/null
+		[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${tmp_file}"
 	}
 	file_or_dir "package.env" "Checking defined configs in" "package_env_def" || return 1
 }
@@ -758,7 +739,7 @@ env_not_installed(){
 			while read category pn ver rev slot;do
 				[[ "${category}" == "(null)" || "${pn}" == "(null)" ]] && continue
 				[[ "${ver}" == "(null)" ]] && ver="${ver//(null)/}"
-				if [[ -n "${rev}" ]] && ! `grep -q \: <<< "${rev}"`;then
+				if [[ -n "${rev}" ]] && ! grep -q \: <<< "${rev}";then
 					full_ver="${ver}-${rev}"
 				else
 					full_ver="${ver}"
@@ -771,7 +752,6 @@ env_not_installed(){
 					if ! ls -d1 ${PKGDB}/${category}/${pn}-[0-9]* &>/dev/null;then
 						echo -e "${red}Not installed${restore} ${green}${category}/${pn}${restore}"
 						RM+="${PORT_ETC}/${x}/${file}%"
-						eend $? 2>/dev/null
 					fi
 				fi
 
@@ -779,7 +759,6 @@ env_not_installed(){
 					if ! ls -d1 ${PKGDB}/${category}/${pn}-${full_ver} &>/dev/null;then
 						echo -e "${red}Not installed${restore} ${green}${category}/${pn}-${full_ver}${restore}"
 						RM+="${PORT_ETC}/${x}/${file}%"
-						eend $? 2>/dev/null
 					fi
 				fi
 
@@ -787,7 +766,6 @@ env_not_installed(){
 					if [[ -z "$(cat ${PKGDB}/${category}/${pn}-[0-9]*/SLOT 2>/dev/null | grep ${slot})" ]];then
 						echo -e "${red}Not installed${restore} ${green}${category}/${pn}:${slot}${restore}"
 						RM+="${PORT_ETC}/${x}/${file}%"
-						eend $? 2>/dev/null
 					fi
 				fi
 			done <<< "${atom}"
@@ -799,11 +777,10 @@ env_not_installed(){
 		done
 		IFS=$'\n'
 		for target in ${dirs};do
-			[[ -z "`ls ${target}`" ]] && remove_ask "${target}"
+			[[ -z "$(ls ${target})" ]] && remove_ask "${target}"
 		done
 	unset IFS
 	done
-	eend $? 2>/dev/null
 }
 ############################################################################
 ################ incorrect uses ############################################
@@ -811,7 +788,7 @@ invalid_uses(){
 	set -f
 	FULL="${PROFILE} ${MAKE_USES}"
 	for opt in ${FULL};do
-		GLOBAL+=" $(echo ${FULL} | tr "[:space:]" $'\n' | grep -e "^${opt#-}$" -e "^\-${opt#-}$" | tail -n1)"
+		GLOBAL+=" $(echo ${FULL} | tr "[:space:]" $'\n' | grep -e "^${opt#-}$" -e "^-${opt#-}$" | tail -n1)"
 	done
 	GLOBAL="$(echo ${GLOBAL} | tr "[:space:]" $'\n' | sort -u)"
 ############################################################################
@@ -838,7 +815,6 @@ invalid_uses(){
 					usefxd="$(sed -e 's| - | |g' -e 's/[ -]*$//' <<< ${usefxd})"
 					strfxd="${category}/${pn} ${usefxd}"
 					sed -e "s|^${category}/${pn}[[:space:]].*${trash#-}.*|$strfxd|" -i ${tmp_file} 2>/dev/null
-					eend $? 2>/dev/null
 				done <<< "${strings}"
 			}
 
@@ -851,7 +827,6 @@ invalid_uses(){
 					usefxd="$(sed -e 's| - | |g' -e 's/[ -]*$//' <<< ${usefxd})"
 					strfxd="${category}/${pn} ${usefxd}"
 					sed -e "s|^${category}/${pn}[[:space:]].*${use#-}.*|$strfxd|" -i ${tmp_file} 2>/dev/null
-					eend $? 2>/dev/null
 				done <<< "${strings}"
 			}
 
@@ -883,7 +858,7 @@ invalid_uses(){
 
 				if grep -q "^${use}$" <<< "${GLOBAL}" && grep -q "^${use}$" <<< "${LOCAL}";then
 					global_use="${global_use} ${use}"
-				elif grep -q "^\-${use}$" <<< "${GLOBAL}" && grep -q "^\-${use}$" <<< "${LOCAL}";then
+				elif grep -q "^-${use}$" <<< "${GLOBAL}" && grep -q "^-${use}$" <<< "${LOCAL}";then
 					global_use="${global_use} -${use}"
 					incorrect_use="${incorrect_use// ${use}/}"
 				fi
@@ -906,27 +881,24 @@ invalid_uses(){
 				done <<< "${global_use}"
 			fi
 			unset ALL_USE LOCAL USES incorrect_use global_use use
-		done <<< "`[[ -n "${atoms}" ]] && echo -e "${atoms}"`"
+		done <<< "$([[ -n "${atoms}" ]] && echo -e "${atoms}")"
 		while read atom opt;do
 			if [[ -n "${atom}" ]] && [[ -z "${opt}" ]];then
 				echo -e "Removing atom: ${green}${atom}${restore}"
 				sed -e "s|^${atom} *$||" -e '/^$/d' -i "${tmp_file}" 2>/dev/null
-				eend $? 2>/dev/null
 			fi
 		done <<< "$([[ -n "${IGNORE}" ]] && grep -v "${IGNORE}" < "${file}" || cat "${file}")"
 		while read atom opt;do
 			if [[ -n "${atom}" ]] && [[ -z "${opt}" ]];then
 				echo -e "Removing atom: ${green}${atom}${restore}"
 				sed -e "s|^${atom} *$||" -e '/^$/d' -i "${tmp_file}" 2>/dev/null
-				eend $? 2>/dev/null
 			fi
 		done <<< "$([[ -n "${IGNORE}" ]] && grep -v "${IGNORE}" < "${tmp_file}" || cat "${tmp_file}")"
-		eend $? 2>/dev/null
 		unset FORMAT USEONLY
 	}
 	if [[ -f "${PORT_ETC}/package.use" ]];then
 		file="${PORT_ETC}/package.use"
-		tmp_file="$(mktemp)"
+		tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 		cp "${file}" "${tmp_file}"
 		if [[ -n "${IGNORE}" ]];then
 			content="$(awk '{print $1}' ${file} | tr -d "<>=~" | sed -e 's|^#.*||g' | grep -v "${IGNORE}")"
@@ -941,13 +913,13 @@ invalid_uses(){
 		check_uses;
 		tput el;
 		tput cnorm;
-		diff_ask "${file}" "${tmp_file}"
+		[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${tmp_file}"
 		[[ ! -s "${file}" ]] && remove_ask "${file}"
 	elif [[ -d "${PORT_ETC}/package.use" ]];then
-		for f in `find ${PORT_ETC}/package.use/ -type f \! -name '*~' \! -name '*.bak'`;do
+		for f in $(find ${PORT_ETC}/package.use/ -type f \! -name '*~' \! -name '*.bak');do
 			if [[ -s "${f}" ]];then
 				file="${f}"
-				tmp_file="$(mktemp)"
+				tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 				cp "${file}" "${tmp_file}"
 				if [[ -n "${IGNORE}" ]];then
 					content="$(awk '{print $1}' ${file} | tr -d "<>=~" | sed -e 's|^#.*||g' | grep -v "${IGNORE}")"
@@ -961,12 +933,11 @@ invalid_uses(){
 				chars="$(wc -m <<< "${count}")"
 				check_uses;
 				tput cnorm
-				diff_ask "${file}" "${tmp_file}"
+				[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${tmp_file}"
 				[[ ! -s "${file}" ]] && remove_ask "${file}"
 			else
 				echo -e "${underline}Removing empty file${restore} ${contrast}${f}${restore}"
 				rm ${f} 2>/dev/null
-				eend $? 2>/dev/null
 				continue
 			fi
 		done
@@ -990,7 +961,7 @@ hell_parser(){
 invalid_uses_make(){
 	set -f
 	echo -e "${underline}Checking use flags in${restore} ${contrast}${makefile}${restore}"
-	tmp_file="$(mktemp)"
+	tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 	cp "${makefile}" "${tmp_file}"
 
 	vars_in_USE="$(make_conf_use USE | hell_parser | grep '\$' | tr -d '${}')"
@@ -1012,7 +983,7 @@ invalid_uses_make(){
 			sed -e "s|${line%\\}|${fixed_line%\\}|" -i "${tmp_file}"
 		done
 	fi
-	diff_ask "${makefile}" "${tmp_file}"
+	[[-z "${NO_DIFF}"]] && diff_ask "${makefile}" "${tmp_file}"
 	set +f
 }
 ############################################################################
@@ -1024,27 +995,25 @@ not_found(){
 		incorrect="$(echo -e "${TRASH}" | tr $'\n' "%" | sed -e "s|.*${package}:||" -e 's|--.*||' -e 's|No.*||'|\
 		tr "%" $'\n' | sed -e '/^$/d' | grep -v "${IGNORE}")"
 		if [[ -n "${incorrect}" && -f "${package}" ]];then
-			tmp_package="$(mktemp)"
+			tmp_package="${TMPDIR_PORTCONF}/tmp_package"
 			cp "${package}" "${tmp_package}"
 			echo -e "${underline}Removing incorrect from${restore} ${contrast}${package}${restore}:"
 			while read line;do
 				echo -e "${red}Incorrect:${restore} ${line}"
 				line="${line//'*'/\*}"
 				[[ -n "${line}" ]] && sed -e "s|$line||" -e 's/[ \t]*$//' -e '/^$/d' -i ${tmp_package} 2>/dev/null
-				eend $? 2>/dev/null
 			done <<< "${incorrect}"
-			diff_ask "${package}" "${tmp_package}"
+			[[-z "${NO_DIFF}"]] && diff_ask "${package}" "${tmp_package}"
 		elif [[ -n "${incorrect}" && -d "${package}" ]];then
 			echo -e "${underline}Removing incorrect from${restore} ${contrast}${package}${restore}:"
 			IFS=$'\n'
 			for line in ${incorrect};do
-				for file in `grep -RH "${line//'*'/\*}" "${package}"/ | awk -F \: '{print $1}'`;do
-					tmp_file="$(mktemp)"
+				for file in $(grep -RH "${line//'*'/\*}" "${package}"/ | awk -F \: '{print $1}');do
+					tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 					cp "${file}" "${tmp_file}"
 					echo -e "${red}Incorrect:${restore} ${line}"
 					[[ -n "${line}" ]] && sed -e "s|${line//'*'/\*}||" -e 's/[ \t]*$//' -e '/^$/d' -i ${tmp_file}
-					diff_ask "${file}" "${tmp_file}"
-					eend $? 2>/dev/null
+					[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${tmp_file}"
 				done
 			done
 			unset IFS
@@ -1068,21 +1037,20 @@ not_found(){
 		INVALID_FOUNDED+="${line}"$'\n'
 
 		if [[ -n "${line}" && -f "${file}" ]];then
-			tmp_file="$(mktemp)"
+			tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 			cp "${file}" "${tmp_file}"
-			[[ -n "${line}" ]] && sed -e "s|`echo $line`$||" -e 's|[ \t]| |' -e '/^[ \t]$/d' -e '/^$/d' -i ${tmp_file}
-			diff_ask "${file}" "${tmp_file}"
+			[[ -n "${line}" ]] && sed -e "s|$(echo "$line")$||" -e 's|[ \t]| |' -e '/^[ \t]$/d' -e '/^$/d' -i "${tmp_file}"
+			[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${tmp_file}"
 		elif [[ -n "${line}" && -d "${file}" ]];then
-			for package in `grep -RH "${line//'*'/\*}" "${file}"/ | awk -F \: '{print $1}'`;do
-					tmp_package="$(mktemp)"
+			for package in $(grep -RH "${line//'*'/\*}" "${file}"/ | awk -F \: '{print $1}');do
+					tmp_package="${TMPDIR_PORTCONF}/tmp_package"
 					cp "${package}" "${tmp_package}"
 					echo -e "${red}Incorrect:${restore} ${line}"
 					[[ -n "${line}" ]] && sed -e "s|${line//'*'/\*}||" -e 's/[ \t]*$//' -e '/^$/d' -i ${tmp_package}
-					diff_ask "${package}" "${tmp_package}"
+					[[-z "${NO_DIFF}"]] && diff_ask "${package}" "${tmp_package}"
 					eend $? 2>/dev/null
 			done
 		fi
-		eend $? 2>/dev/null
 	done
 	unset IFS
 
@@ -1095,14 +1063,14 @@ not_found(){
 	fi
 	broken_files="$(awk '{print $1}' <<< "${invalid}" | sort -u)"
 	invalid_lines(){
-		tmp_file="$(mktemp)"
+		tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 		cp "${file}" "${tmp_file}"
 		set -f
 		tlines="$(grep "^${file} " <<< "${invalid}" | sed "s|${file} ||g")"
 		while read line;do
-			sed -e "s|^`printf ${line}`.*$||;s|^${line//\*/\\*}.*$||" -e 's|[ \t]| |;/^[ \t]$/d;/^$/d' -i "${tmp_file}"
+			sed -e "s|^$(printf ${line}).*$||;s|^${line//\*/\\*}.*$||" -e 's|[ \t]| |;/^[ \t]$/d;/^$/d' -i "${tmp_file}"
 		done <<< "${tlines}"
-		diff_ask "${file}" "${tmp_file}"
+		[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${tmp_file}"
 		set +f
 	}
 	if [[ -n "${invalid}" ]];then
@@ -1118,7 +1086,7 @@ use_makeconf(){
 	echo -e "${underline}Checking use flags in${restore} ${contrast}${makefile##/etc*/}${restore}"
 	USE="$(echo -e "${MAKE_USES}" | tr "[:space:]" $'\n' | sort -u | sed -e '/^$/d')"
 
-	tmp_file="$(mktemp)"
+	tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 	cp "${makefile}" "${tmp_file}"
 	while read use;do
 		flag="${use#-}"
@@ -1126,32 +1094,29 @@ use_makeconf(){
 			if grep -q "^${flag}$" <<< "${USE}" && grep -q "^${flag}$" <<< "${PROFILE}";then
 				echo -e "Twice defined: ${red}${use}${restore}"
 				sed -e "s|[^[:punct:]]\<${flag}\>||g" -i ${tmp_file}
-			elif grep -q "^\-${flag}$" <<< "${USE}" && grep -q "^\-${flag}$" <<< "${PROFILE}";then
+			elif grep -q "^-${flag}$" <<< "${USE}" && grep -q "^-${flag}$" <<< "${PROFILE}";then
 				echo -e "Twice defined: ${red}-${use}${restore}"
 				sed -e "s|[^[:punct:]]\-\<${flag}\>||g" -i ${tmp_file}
 			fi
 		fi
 	done <<< "${PROFILE}"
-	diff_ask "${makefile}" "${tmp_file}"
-	eend $? 2>/dev/null
+	[[-z "${NO_DIFF}"]] && diff_ask "${makefile}" "${tmp_file}"
 }
 ############################################################################
 ################ empty files ###############################################
 empty_files(){
 	echo -e "${underline}Looking empty files${restore}"
-	for target in `find ${PORT_ETC} -type f`;do
-		[[ ! -s "${target}" || -z "$(grep -v "^#" < "${target}")" ]] && remove_ask "${target}"
+	for target in $(find ${PORT_ETC} -type f);do
+		[[ ! -s "${target}" || -z "$(awk '!/^#/' < "${target}")" ]] && remove_ask "${target}"
 	done
-	eend 0 2>/dev/null
 }
 ############################################################################
 ################ backup files ##############################################
 backup_files(){
 	echo -e "${underline}Looking backup files${restore}"
-	for target in `find ${PORT_ETC} -type f \( -name "*~" -o  -name "*.bak" -o -name "*.old" \)`;do
+	for target in $(find ${PORT_ETC} -type f \( -name "*~" -o  -name "*.bak" -o -name "*.old" \));do
 		remove_ask "${target}"
 	done
-	eend 0 2>/dev/null
 }
 ############################################################################
 ################ package.* files 2 dirs ####################################
@@ -1159,15 +1124,15 @@ f_to_d(){
 	while read package;do
 		if [[ -f "${package}" ]];then
 			echo -e "${underline}Processing${restore}: ${contrast}${package##*/}${restore}"
-			tmp_file="$(mktemp)"
-			mv ${package} ${tmp_file} 2>/dev/null
+			tmp_file="${TMPDIR_PORTCONF}/tmp_file"
+			mv "${package}" "${tmp_file}" 2>/dev/null
 			sed -e 's|^#.*||g' -i "${tmp_file}"
 			dir="${PORT_ETC}/${package##*/}"
 			mkdir "${dir}"
 			while read line;do
 			set -f
 				atom="$(awk '{print $1}' <<< "${line}")"
-				category="$(awk '{print $1}' <<< `qatom ${atom}`)"
+				category="$(awk '{print $1}' <<< $(qatom ${atom}))"
 				if [[ "${category}" != "(null)" ]];then
 					category="$(tr -d "<>=~" <<< "${category}" )"
 					[[ "${category}" == "*" ]] && category="all_categories"
@@ -1175,13 +1140,12 @@ f_to_d(){
 				fi
 			done < "${tmp_file}"
 			set +f
-			eend $? 2>/dev/null
 			rm "${tmp_file}" 2>/dev/null
 		elif [[ -d "${package}" ]];then
 			echo -e "${underline}Skipping ${red}directory${restore}: ${contrast}${package##*/}${restore}"
 			eend $? 2>/dev/null
 		fi
-	done <<< "`ls -d1 ${PORT_ETC}/package* | grep -v "~"`"
+	done <<< "$(ls -d1 ${PORT_ETC}/package* | grep -v "~")"
 }
 ############################################################################
 ################ package.* dirs 2 files ####################################
@@ -1189,10 +1153,10 @@ d_to_f(){
 	while read package;do
 		if [[ -d "${package}" ]];then
 			echo -e "${underline}Processing${restore}: ${contrast}${package##*/}${restore}"
-			tmp_file="$(mktemp)"
+			tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 			file="${PORT_ETC}/${package##*/}"
 			touch "${file}"
-			for subfile in `find "${package}" -type f \! -name '*~' \! -name '*.bak' | sort`;do
+			for subfile in $(find "${package}" -type f \! -name '*~' \! -name '*.bak' | sort);do
 				while read line;do
 					echo "${line}" >> "${tmp_file}"
 				done < "${subfile}"
@@ -1200,13 +1164,12 @@ d_to_f(){
 			rm -rf "${package}" 2>/dev/null
 			mv "${tmp_file}" "${package}" 2>/dev/null
 			chmod 0644 "${package}"
-			eend $? 2>/dev/null
 			rm "${tmp_file}" 2>/dev/null
 		elif [[ -f "${package}" ]];then
 			echo -e "${underline}Skipping ${red}file${restore}: ${contrast}${package##*/}${restore}"
 			eend $? 2>/dev/null
 		fi
-	done <<< "`ls -d1 ${PORT_ETC}/package* | grep -v "~"`"
+	done <<< "$(ls -d1 ${PORT_ETC}/package* | grep -v "~")"
 }
 ############################################################################
 ################ version diff ##############################################
@@ -1240,19 +1203,19 @@ package_envs(){
 			slot="${slot#:}"
 			pn="${pn%:}"
 			echo "${sw} ${category} ${pn} ${ver} ${rev:-(null)} ${slot:-(null)} ${file}"
-		done <<< `qatom "${atom}"`
+		done <<< $(qatom "${atom}")
 	done
 }
 ############################################################################
 ################ mask/unmask trash #########################################
 all_mask_trash(){
 	pre_mask_trash(){
-		masked="$(awk '{print $1}' ${file} | grep -v '^#.*')"
+		masked="$(awk '!/^#/ {print $1}' ${file})"
 		MASKED+="$(package_envs "${masked}")"$'\n'
 	}
 	file_or_dir "package.mask" "Preparing atoms in" "pre_mask_trash" || return 1
 	pre_unmask_trash(){
-		unmasked="$(awk '{print $1}' ${file} | grep -v '^#.*')"
+		unmasked="$(awk '!/^#/ {print $1}' ${file})"
 		UNMASKED+="$(package_envs "${unmasked}")"$'\n'
 	}
 	file_or_dir "package.unmask" "Preparing atoms in" "pre_unmask_trash" || return 1
@@ -1333,13 +1296,13 @@ mask_trash(){
 	files="$(awk '{print $1}' <<< "${TRASH}" | sort -u | sed '/^$/d')"
 	for target in ${files};do
 		if [[ -f "${target}" ]];then
-			tmp_file="$(mktemp)"
+			tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 			cp "${target}" "${tmp_file}"
 			incorrect="$(grep ${target} <<< "${TRASH}" | sed -e "s|${target} ||g")"
 			for trash_atom in ${incorrect};do
 				sed -e "s|^${trash_atom}.*||" -e '/^$/d' -e 's/[ \t]*$//' -i "${tmp_file}"
 			done
-			diff_ask "${target}" "${tmp_file}"
+			[[-z "${NO_DIFF}"]] && diff_ask "${target}" "${tmp_file}"
 		fi
 	done
 	unset TRASH OVERLAYS_LIST VERSIONONLY FORMAT MASKED UNMASKED versions oldest versions_diff versions_btwn
@@ -1347,15 +1310,15 @@ mask_trash(){
 ############################################################################
 ################ looking trash #############################################
 remove_trash(){
-	tmp_file="$(mktemp)"
+	tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 	cp "${file}" "${tmp_file}"
-	for atom in `eix -Tc#`;do
+	for atom in $(eix -Tc#);do
 		lines="$(grep "${atom}" "${file}")"
 		while read line;do
 			entry="$(qatom "$(tr -d '<>=~#' <<< "${line}")" | awk '{print $1,$2}' | tr ' ' '/')"
 			if [[ "${atom}" == "${entry}" ]];then
 				if ! eix -qe --installed-masked "${entry}";then
-					if `eix -Iqe ${atom}`;then
+					if $(eix -Iqe "${atom}");then
 						echo -e "stupid entry:              ${green}${line}${restore}"
 					else
 						echo -e "stupid entry (${red}!${restore}installed): ${green}${line}${restore}"
@@ -1366,8 +1329,7 @@ remove_trash(){
 			fi
 		done <<< "${lines}"
 	done
-	diff_ask "${file}" "${tmp_file}"
-	eend $? 2>/dev/null
+	[[-z "${NO_DIFF}"]] && diff_ask "${file}" "${tmp_file}"
 }
 ############################################################################
 ################ stupid unmask #############################################
@@ -1383,16 +1345,15 @@ stupid_unmask(){
 ################ stupid mask/unmask uses ###################################
 sort_uniq_mask_use(){
 	if [[ -d "${PORT_ETC}/profile" || -h "${PORT_ETC}/profile" ]];then
-		for package in `find ${PORT_ETC}/profile -type f \! -name '*~' \! -name '*.bak' -name "use.stable.mask" -or -name "use.mask"`;do
-			tmp_sort="$(mktemp)"
+		for package in $(find ${PORT_ETC}/profile -type f '(' \! -name '*~' \! -name '*.bak' ')' '(' -name "use.stable.mask" -o -name "use.mask" ')');do
+			tmp_sort="${TMPDIR_PORTCONF}/tmp_sort"
 			echo -e "${underline}Sorting${restore} ${contrast}${package#${PORT_ETC}/}${restore}"
 
-			masked_uses="$(grep -v "^#" "${package}")"
+			masked_uses="$(awk '!/^#/' "${package}")"
 
 			sort_passed_uses "${masked_uses}" | tr "[:space:]" $'\n' > ${tmp_sort}
 
-			diff_ask "${package}" "${tmp_sort}"
-			eend $? 2>/dev/null
+			[[-z "${NO_DIFF}"]] && diff_ask "${package}" "${tmp_sort}"
 		done
 	fi
 }
@@ -1432,10 +1393,10 @@ etc_profile_use_mask() {
 	else
 		return 1
 	fi
-	tmp_file="$(mktemp)"
+	tmp_file="${TMPDIR_PORTCONF}/tmp_file"
 	cp "${orig_file}" "${tmp_file}"
 	while read use;do
-		if grep -q "^${use/-/\-}$" <<< "$(use_mask)";then
+		if grep -q "^${use/-/-}$" <<< "$(use_mask)";then
 			if grep -q "^-" <<< "${use}";then
 				echo -e "Twice unmasked: ${red}${use}${restore}"
 				sed -e "s|^${use}.*||" -e '/^$/d' -e 's/[ \t]*$//' -i "${tmp_file}"
@@ -1450,7 +1411,7 @@ etc_profile_use_mask() {
 			fi
 		fi
 	done < "${orig_file}"
-	diff_ask "${orig_file}" "${tmp_file}"
+	[[-z "${NO_DIFF}"]] && diff_ask "${orig_file}" "${tmp_file}"
 }
 
 check_use_masked() {
@@ -1461,9 +1422,9 @@ check_use_masked() {
 ############################################################################
 ################ regen_world ###############################################
 regen_world(){
-	tmp_world="$(mktemp)"
-	ex_world="$(mktemp)"
-	emerge_log="$(mktemp)"
+	tmp_world="${TMPDIR_PORTCONF}/tmp_world"
+	ex_world="${TMPDIR_PORTCONF}/ex_world"
+	emerge_log="${TMPDIR_PORTCONF}/emerge_log"
 	mv "${WORLD}" "${ex_world}"
 	installed="$(qlist -CI | sort -u)"
 	echo "${installed}" > "${WORLD}"
@@ -1482,7 +1443,7 @@ regen_world(){
 	fi
 
 	pretend="$(grep -v "blocks" <<< "${pre_pre}" | awk '/^\[/ {print $4}')" && eend 0 2>/dev/null
-	pretend+="$(qlist -CI `emerge -epO system | awk '/\[/ {print $4}'`)"$'\n'
+	pretend+="$(qlist -CI $(emerge -epO system | awk '/\[/ {print $4}'))"$'\n'
 
 	for atom in ${installed};do
 		if ! grep -q "^${atom}$" <<< "${pretend}" && ! grep -q ".*-libs/" <<< "${atom}" && ! grep -q "virtual/" <<< "${atom}";then
@@ -1492,7 +1453,7 @@ regen_world(){
 	done
 	cp "${tmp_world}" "${WORLD}"
 
-	PS3=`echo -e "${contrast}Choose package which you want save:${restore} "`
+	PS3=$(echo -e "${contrast}Choose package which you want save:${restore} ")
 	choose_target(){
 		select target in ${pretend};do
 			echo "=${target}"
@@ -1550,9 +1511,8 @@ regen_world(){
 	done
 	cp "${WORLD}" "${tmp_world}"
 	cp "${ex_world}" "${WORLD}"
-	diff_ask "${WORLD}" "${tmp_world}"
+	[[-z "${NO_DIFF}"]] && diff_ask "${WORLD}" "${tmp_world}"
 	rm "${ex_world}"
-	eend $? 2>/dev/null
 	echo -e "\n${green}Result${restore}:"
 	cat "${WORLD}"
 }
@@ -1604,7 +1564,7 @@ overlays(){
 		repo_path="$(grep " ${repo}$" <<< "${portage_overlays}" | awk '{print $1}')"
 		if [[ -d "${repo_path}" ]];then
 			######## checking symlinks in repo
-			for x in `find "${repo_path}" -type l`;do
+			for x in $(find "${repo_path}" -type l);do
 				target="$(readlink ${x})"
 				if [[ -e "${target}" || -e "${x%/*}/${target}" ]];then
 					good_links+="${target}"$'\n'
@@ -1644,7 +1604,6 @@ overlays(){
 				Yes | YES | yes | Y | y | "" )
 					for symlink in $(awk '{print $2}' <<< "${bad_links%$'\n'}");do
 						rm -v "${symlink}"
-						eend $? 2>/dev/null
 					done
 					break;;
 				No | NO | no | N | n )
@@ -1656,7 +1615,7 @@ overlays(){
 	fi
 
 	choose_target(){
-		PS3=`echo -e "${contrast}Choose repo which you want save:${restore} "`
+		PS3=$(echo -e "${contrast}Choose repo which you want save:${restore} ")
 		select target in ${UNUSED};do
 			echo "${target}"
 			break
@@ -1700,7 +1659,7 @@ overlays(){
 	fix_deps(){
 		DEP_PATH="/var/cache/edb/dep"
 		unset TRASH
-		for dep_path in `find ${DEP_PATH} -type d`;do
+		for dep_path in $(find ${DEP_PATH} -type d);do
 			repo_path="$(sed "s|${DEP_PATH}||" <<< "${dep_path}")"
 			dep_file="${dep_path}.sqlite"
 			if [[ -n "${repo_path}" ]];then
@@ -1739,14 +1698,12 @@ overlays(){
 		trash="$(fix_deps)"
 		if [[ -n "${trash}" ]];then
 			echo -e "$(tr ' ' $'\n' <<< "${trash% }" | sed "s|^|removing:\\${green}|;s|$|\\${restore}|")"
-			rm -rf ${trash} 2>/dev/null || echo failed && exit 1
+			rm -rf "${trash}" 2>/dev/null || echo failed && exit 1
 		else
 			break
-			
 		fi
 	done
 	echo "done"
-	eend 0 2>/dev/null
 }
 ############################################################################
 ################ help ######################################################
@@ -1762,6 +1719,8 @@ ${contrast}Common${restore}
 	-s,  --sort:                 Sort entries alphabetically, --use-sort, --keyword-uniq.
 	-y,  --yes:                  Auto answer (Yes).
 	-p,  --pretend:              Check only.
+	--no-backup:               Skip backup before operations.
+	--no-diff:                 Skip diff prompts, apply changes directly.
 	-h,  -h, -?, h, --help:      This help.
 ${contrast}\${USE}${restore}
 	-us, --use-sort:             Sort useflags. Remove doubles and preserve only last defined state (on/off).
@@ -1811,18 +1770,20 @@ eix_dep_keys="--use-invalid
 --full
 -f"
 opts=" ${PORTCONF_DEFAULT_OPTS} ${@}"
-grep -q -e " \--yes\>" -e " \-y\>" <<< " ${opts}" && yes="1"
-grep -q -e " \--pretend\>" -e " \-p\>" <<< " ${opts}" && PRETEND="1"
-opts="$(sed -e 's| --yes\>||g;s| -y\>||g' -e 's| --pretend\>||g;s| -p\>||g' <<< " ${opts}")"
+grep -q -e " --yes\>" -e " -y\>" <<< " ${opts}" && yes="1"
+grep -q -e " --pretend\>" -e " -p\>" <<< " ${opts}" && PRETEND="1"
+grep -q -e " --no-backup\>" <<< " ${opts}" && NO_BACKUP="1"
+grep -q -e " --no-diff\>" <<< " ${opts}" && NO_DIFF="1"
+opts="$(sed -e 's| --yes\>||g;s| -y\>||g' -e 's| --pretend\>||g;s| -p\>||g' -e 's| --no-backup\>||g' -e 's| --no-diff\>||g' <<< " ${opts}")"
 if grep -q "${eix_dep_keys//\-/\\-}" <<< "${opts}";then
-	if grep -q -e " \--regen-cache\>" -e " \-rc\>" <<< " ${opts}";then
+	if grep -q -e " --regen-cache\>" -e " -rc\>" <<< " ${opts}";then
 		eix_check
-		opts="$(sed -e 's| --regen-cache\>||g;s| -rc\>||g' -e '/^[ \t]$/d' <<< " ${opts}")"
+		opts="$(sed -e 's| --regen-cache\>||g;s| -rc\>||g' -e 's| --no-backup\>||g' -e 's| --no-diff\>||g' -e '/^[ \t]$/d' <<< " ${opts}")"
 	else
 		eix_method
 	fi
 else
-	opts="$(sed -e 's| --regen-cache\>||g;s| -rc\>||g' -e '/^[ \t]$/d' <<< " ${opts}")"
+	opts="$(sed -e 's| --regen-cache\>||g;s| -rc\>||g' -e 's| --no-backup\>||g' -e 's| --no-diff\>||g' -e '/^[ \t]$/d' <<< " ${opts}")"
 fi
 [[ -z "${@}" || -z "${opts}" ]] && help_pc
 for OPT in ${opts};do
@@ -1830,28 +1791,28 @@ for OPT in ${opts};do
 	--backup | -b ) backup;;
 	--restore | -r ) etc_restore;;
 	--sort | -s )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		sort_use_file;
 		uniq_keywords;
 		sort_uniq_files;
 		sort_uniq_mask_use;;
 	--use-sort | -us )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		sort_use_file;;
 	--use-invalid | -ui )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		invalid_uses;
 		invalid_uses_make;
 		[[ -n "${eix_cache}" ]] && rm "${eix_cache}" && unset eix_cache EIX_CACHEFILE;;
 	--use-make | -um )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		use_makeconf;;
 	--stupid-use-profile | -sup )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		sort_uniq_mask_use;
 		check_use_masked;;
 	--use-full | -uf )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		sort_use_file;
 		sort_uniq_mask_use;
 		invalid_uses;
@@ -1860,13 +1821,13 @@ for OPT in ${opts};do
 		check_use_masked;
 		[[ -n "${eix_cache}" ]] && rm "${eix_cache}" && unset eix_cache EIX_CACHEFILE;;
 	--keyword-one | -ko )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		sort_keywords;;
 	--keyword-uniq | -ku )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		uniq_keywords;;
 	--trash | -t )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		not_found;
 		package_env_conf;
 		package_env;
@@ -1875,13 +1836,13 @@ for OPT in ${opts};do
 		backup_files;
 		[[ -n "${eix_cache}" ]] && rm "${eix_cache}" && unset eix_cache EIX_CACHEFILE;;
 	--stupid-mask | -sm )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		mask_trash;;
 	--stupid-unmask | -sum )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		stupid_unmask;;
 	--force-trash | -ft )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		not_found;
 		package_env_conf;
 		package_env;
@@ -1891,13 +1852,13 @@ for OPT in ${opts};do
 		backup_files;
 		[[ -n "${eix_cache}" ]] && rm "${eix_cache}" && unset eix_cache EIX_CACHEFILE;;
 	--rm-comments | -c )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		rm_comments;;
 	--rm-all-comments | -ac )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		rm_all_comments;;
 	--full | -f )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		sort_uniq_files;
 		not_found;
 		package_env_conf;
@@ -1916,10 +1877,10 @@ for OPT in ${opts};do
 		backup_files;
 		[[ -n "${eix_cache}" ]] && rm "${eix_cache}" && unset eix_cache EIX_CACHEFILE;;
 	--files-2-dirs | -f2d )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		f_to_d;;
 	--dirs-2-files | -d2f )
-		backup;
+		[[ -z "${NO_BACKUP}" ]] && backup;
 		d_to_f;;
 	--all-profiles-use | -apu ) all_use_all_profiles;;
 	--profiles-use | -pu ) profile_use;;
@@ -1934,3 +1895,6 @@ for OPT in ${opts};do
 	--help | -h | -? | h ) help_pc;;
 	esac
 done
+############################################################################
+################ cleanup #################################################
+rm -rf "${TMPDIR_PORTCONF}" 2>/dev/null
