diff --git a/bsnes/target-ui/general/main-window.cpp b/bsnes/target-ui/general/main-window.cpp
index a109cbd..3f6382f 100755
--- a/bsnes/target-ui/general/main-window.cpp
+++ b/bsnes/target-ui/general/main-window.cpp
@@ -331,6 +331,10 @@ void MainWindow::setupVideoFilters() {
     path = { application->userpath, "filters/" };
     files = directory::files(path, "*.filter");
   }
+  if(files.size() == 0) {
+    path = "%GAMES_DATADIR%/bsnes/filters/";
+    files = directory::files(path, "*.filter");
+  }
   array<RadioItem&> group;
 
   settingsVideoFilterList = new RadioItem[files.size()];
@@ -362,6 +366,10 @@ void MainWindow::setupVideoShaders() {
     path = { application->userpath, "shaders/" };
     files = directory::files(path, { "*.", config->video.driver, ".shader" });
   }
+  if(files.size() == 0) {
+    path = "%GAMES_DATADIR%/bsnes/shaders/";
+    files = directory::files(path, { "*.", config->video.driver, ".shader" });
+  }
   array<RadioItem&> group;
 
   settingsVideoShaderList = new RadioItem[files.size()];
diff --git a/bsnes/target-ui/tools/cheat-database.cpp b/bsnes/target-ui/tools/cheat-database.cpp
index 9b71a51..bf95610 100755
--- a/bsnes/target-ui/tools/cheat-database.cpp
+++ b/bsnes/target-ui/tools/cheat-database.cpp
@@ -34,7 +34,8 @@ void CheatDatabase::findCodes() {
   cheatCode.reset();
 
   string data;
-  data.readfile(application->path("cheats.xml"));
+  if(data.readfile(application->path("cheats.xml")) == false)
+    data.readfile("%GAMES_DATADIR%/bsnes/cheats.xml");
   XML::Document document(data);
   for(auto &node : document["database"]) {
     if(node.name != "cartridge") continue;
