--- a/src/libgrooveshark/config.cpp
+++ b/src/libgrooveshark/config.cpp
@@ -10,7 +10,7 @@
 using namespace GrooveShark;
 
 Config* Config::s_instance = 0;
 
-ConfigPrivate::ConfigPrivate( Config* qq ) : q( qq ), m_gsBaseUrl( QUrl( QLatin1String( "https://grooveshark.com" ) ) ), m_userAgentPrefix( QString() )
+ConfigPrivate::ConfigPrivate( Config* qq ) : q( qq ), m_gsBaseUrl( QUrl( QLatin1String( "https://groovesharks.org" ) ) ), m_userAgentPrefix( QString() )
 {
 
 }
@@ -37,7 +37,7 @@
 QByteArray ConfigPrivate::userAgent() const
 {
 
 QLatin1String ConfigPrivate::host() const
 {
-    return QLatin1String("grooveshark.com");
+    return QLatin1String("groovesharks.org");
 }
 
 QString ConfigPrivate::userAgentPrefix() const
